language/move-prover/boogie-backend/src/boogie_wrapper.rs

```diff
@@ -575,21 +609,15 @@ pub enum ValueArrayRep {
 #[derive(Debug)]
 pub struct Model {
     vars: BTreeMap<ModelValue, ModelValue>,
-    tracked_exps: BTreeMap<ExpDescriptor, Vec<ModelValue>>,
-    value_array_rep: ValueArrayRep,
+    vector_theory: VectorTheory,
```

Comment:
We might want to pass a `BoogieOptions` here instead of just the `VectorTheory`? This is just a random thought and I understand that there are no dependencies to other fields in `BoogieOptions` except `options.vector_theory`, but we may run into one in the future. The mental model is, the options guides how Boogie code are generated, and hence, might also affect how Boogie results should be parsed.

Reply:
Yes. This was some code created by an intern a while ago which I refactor here. As soon as we need we should put the options directly here.